woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php

```diff
@@ -672,6 +672,39 @@ public function save() {
 	}
 
 
+	/**
+	 * Deletes a token from the database.
+	 *
+	 * Also deletes a token from the legacy user meta.
+	 * @see \WC_Payment_Token::delete()
+	 * @see SV_WC_Payment_Gateway_Payment_Tokens_Handler::delete_legacy_token()
+	 *
+	 * @since 5.6.0-dev.1
+	 *
+	 * @param bool $force_delete argument mapped to {@see \WC_Data::delete()}
+	 * @return bool whether a token was deleted (note: it will not evaluate deletion of legacy data)
+	 */
+	public function delete( $force_delete = false ) {
+
+		$deleted = false;
+
+		if ( $token = $this->get_woocommerce_payment_token() ) {
+			$deleted = $token->delete( $force_delete );
+		}
+
+		// delete legacy token in user meta data
+		$gateways   = WC()->payment_gateways()->payment_gateways();
+		$gateway_id = $this->get_gateway_id();
+
+		if ( ! empty( $gateway_id ) && ! empty( $gateway = $gateways[ $gateway_id ] ) ) {
```

Comment:
we should break this var assignment out of the conditional for readability. Also, should check that the gateway is an instance of `\SV_WC_Payment_Gateway` before calling the tokens handler, just in case.

Reply:
Ok I will have to update both this bit and the one in `save()` since they are quite the same